java/kafka/consumer/src/main/java/KafkaConsumerConfig.java

```diff
@@ -3,228 +3,55 @@
         this.topic = topic;
-        this.groupId = groupId;
-        this.clientRack = clientRack;
         this.messageCount = messageCount;
-        this.sslTruststoreCertificates = sslTruststoreCertificates;
-        this.sslKeystoreKey = sslKeystoreKey;
-        this.sslKeystoreCertificateChain = sslKeystoreCertificateChain;
-        this.oauthClientId = oauthClientId;
-        this.oauthClientSecret = oauthClientSecret;
-        this.oauthAccessToken = oauthAccessToken;
-        this.oauthRefreshToken = oauthRefreshToken;
-        this.oauthTokenEndpointUri = oauthTokenEndpointUri;
-        this.additionalConfig = additionalConfig;
         this.tracingSystem = tracingSystem;
+        this.properties = properties;
     }
 
     public static KafkaConsumerConfig fromEnv() {
-        String bootstrapServers = System.getenv("BOOTSTRAP_SERVERS");
-        String topic = System.getenv("TOPIC");
-        String groupId = System.getenv("GROUP_ID");
-        String clientRack = System.getenv("CLIENT_RACK");
-        Long messageCount = System.getenv("MESSAGE_COUNT") == null ? DEFAULT_MESSAGES_COUNT : Long.parseLong(System.getenv("MESSAGE_COUNT"));
-        String sslTruststoreCertificates = System.getenv("CA_CRT");
-        String sslKeystoreKey = System.getenv("USER_KEY");
-        String sslKeystoreCertificateChain = System.getenv("USER_CRT");
-        String oauthClientId = System.getenv("OAUTH_CLIENT_ID");
-        String oauthClientSecret = System.getenv("OAUTH_CLIENT_SECRET");
-        String oauthAccessToken = System.getenv("OAUTH_ACCESS_TOKEN");
-        String oauthRefreshToken = System.getenv("OAUTH_REFRESH_TOKEN");
-        String oauthTokenEndpointUri = System.getenv("OAUTH_TOKEN_ENDPOINT_URI");
-        String additionalConfig = System.getenv().getOrDefault("ADDITIONAL_CONFIG", "");
-        TracingSystem tracingSystem = TracingSystem.forValue(System.getenv().getOrDefault("TRACING_SYSTEM", ""));
-
-        return new KafkaConsumerConfig(bootstrapServers, topic, groupId, clientRack, messageCount, sslTruststoreCertificates, sslKeystoreKey,
-                sslKeystoreCertificateChain, oauthClientId, oauthClientSecret, oauthAccessToken, oauthRefreshToken, oauthTokenEndpointUri,
-                additionalConfig, tracingSystem);
+        String topic = System.getenv("STRIMZI_TOPIC");
+        Long messageCount = System.getenv("STRIMZI_MESSAGE_COUNT") == null ? DEFAULT_MESSAGES_COUNT : Long.parseLong(System.getenv("STRIMZI_MESSAGE_COUNT"));
+        TracingSystem tracingSystem = TracingSystem.forValue(System.getenv().getOrDefault("STRIMZI_TRACING_SYSTEM", ""));
+        Properties properties = new Properties();
+        properties.putAll(System.getenv()
+                .entrySet()
+                .stream()
+                .filter(map -> map.getKey().startsWith(KAFKA_PREFIX))
+                .collect(Collectors.toMap(map -> convertEnvVarToPropertyKey(map.getKey()), map -> map.getValue())));
```

Comment:
is `map` appropriate as variable here? You are getting a `Map.Entry` instance right? I would be more for naming it `entry` or `mapEntry` but not just `map` which makes reader thinking about the whole map itself.

Reply:
Refactored to use mapEntry